File: typescript/src/PlanetTS1/Model/World/Minerals.ts
import { ALocation as ALocation } from "../Utilities"

export class Minerals extends Resource {
    loc : ALocation
    icon : string
    health : number
    minSal : number
    gasSal : number
    nrgSal : number

    constructor(loc: ALocation) {
        super()
        this.loc = loc
        this.icon = "M"
        this.health = 50
        this.minSal = 10
        this.gasSal = 0
        this.nrgSal = 0
    }
}

File: dotnet/src/Model/World/Minerals.cs
using System;

namespace PlanetCS1 {
    class Minerals : Resource {
        public Minerals(Location location) {
            this.loc = location;
            this.icon = "M";
            this.health = 50;
            this.minSal = 10;
            this.gasSal = 0;
            this.nrgSal = 0;
        }
    }
}